src/core/elf.h

```diff
@@ -40,6 +32,47 @@ typedef struct elf_program_header {
     uint64_t align;
 } __attribute__((packed)) elf_program_header_t;
 
-elf_header_t* elf_load(uint64_t* data);
+typedef struct elf_section_header {
+
+    uint32_t name; //Section Name
+    uint32_t type; //Section Type
+    uint64_t flags; //Section attributes
+    uint64_t addr; //Virtual Address in memory
+    uint64_t offset; //Offset in file
+    uint64_t size; //Size of section
+    uint32_t link; //Link to other section
+    uint32_t info; //Miscellaneous information
+    uint64_t addralign; //Address alignment boundary
+    uint64_t entsize; //Size of entries, if section has table
```

Comment:
If we want to keep these comments, then we need to make them Doxygen compliant. I think this would work:

```suggestion
    uint64_t entsize; ///< Size of entries, if section has table.
```

Reply:
Done